src/pretalx/api/views/event.py

```diff
@@ -24,3 +41,125 @@ def get_object(self):
         if self.request.user.has_perm(self.permission_required, self.request.event):
             return self.request.event
         raise Http404()
+
+
+@api_view(http_method_names=["POST"])
+@authentication_classes([])
+@permission_classes([])
+def configure_video_settings(request):
+    """
+    Configure video settings for an event
+    @param request: request object
+    @return response object
+    """
+    try:
+        video_settings = request.data.get("video_settings")
+
+        if not video_settings or "secret" not in video_settings:
+            raise ValueError("Video settings are missing or secret is not provided")
+
+        payload = get_payload_from_token(request, video_settings)
+        event_slug = payload.get("event_slug")
+        video_tokens = payload.get("video_tokens")
+
+        with scopes_disabled():
+            event_instance = Event.objects.get(slug=event_slug)
+            save_video_settings_information(event_slug, video_tokens, event_instance)
+    except Event.DoesNotExist:
+        logger.error("Event with slug %s does not exist.", event_slug)
+        return Response(
+            {
+                "status": "error",
+                "message": "Event with slug {} not found.".format(event_slug),
+            },
+            status=404,
+        )
+    except ValueError as e:
+        logger.error("Error configuring video settings: %s", e)
+        return Response(
```

Comment:
Why this API endpoint suddenly behave differently from DjangoRestFramework?

If our API is implemented with DjangoRestFramework, we should follow its convention, its style. Don't create inconsistency.

The HTTP status other than 2xx is already indicate error, not need to add `status` field.
In case of validation error (error due to user submitting data in wrong shape), DRF returns errors as a dict like this:

```py
{
   'field_1': ['Error 1', 'Error 2'],
   'field_2': ['Error 1', 'Error 2'],
 }
 ```

Reply:
Hi @hongquan, thanks for the comment.
I implement this REST API which will be called by Video system, it not show the error on template, and it behavior different with others API.
I'm not sure its answered your concern or not, please let me know if you have any else question.